inferno-ml-server/src/Inferno/ML/Server/Bridge.hs

```diff
@@ -29,50 +29,45 @@ import Servant.Client.Streaming
     runClientM,
   )
 import System.Posix.Types (EpochTime)
-import UnliftIO.IORef (atomicWriteIORef, readIORef)
 
--- | Save the provided 'BridgeInfo' and update the Inferno interpreter to use
--- the given bridge (the special primitives must call the bridge to read\/write
--- data from\/to the data source)
-registerBridgeInfo :: BridgeInfo -> RemoteM ()
-registerBridgeInfo bi = do
-  logInfo $ RegisteringBridge bi
-  liftIO $ encodeFile bridgeCache bi
-  (`atomicWriteIORef` Just bi) =<< view (#bridge . #info)
-  interpreter <- mkInferno @_ @BridgeMlValue (mkBridgePrelude funs) customTypes
-  (`atomicWriteIORef` Just interpreter)
-    =<< view #interpreter
+-- | Retrieve the 'BridgeInfo' associated with an inference param and update the
+-- Inferno interpreter to use the given bridge (the special primitives must call
+-- the bridge to read\/write data from\/to the data source)
+initializeInferno ::
+  Id InferenceParam -> RemoteM (Interpreter RemoteM BridgeMlValue)
+initializeInferno ipid = do
+  (`mkInferno` customTypes) . mkBridgePrelude . mkFuns
+    =<< getBridgeInfo
   where
-    funs :: BridgeFuns RemoteM
-    funs = mkBridgeFuns valueAt latestValueAndTimeBefore valuesBetween
-
-    valueAt :: Int64 -> PID -> EpochTime -> RemoteM IValue
-    valueAt res pid = callBridge . Bridge.valueAtC res pid
+    -- There should always be a bridge saved for the param
+    getBridgeInfo :: RemoteM BridgeInfo
+    getBridgeInfo = firstOrThrow NoBridgeSaved =<< queryStore q (Only ipid)
+      where
+        q :: Query
+        q = [sql| SELECT * FROM bridges WHERE id = ? |]
```

Comment:
ohh this syntax `[sql| |]` would be really nice on sqlite, I'm wondering if there is some lib out there.

Reply:
I think there's a raw string quasi-quoter that works with `sqlite-simple`: https://hackage.haskell.org/package/raw-strings-qq